### ISSUE TITLE
123

### ISSUE THREAD

**Issue by SteveHong1901** [SteveHong1901/LSE-Datathon-2022#1]
import statsmodels.api as sm
from sklearn.model_selection import train_test_split

# Assuming your original data is in X and y
X_train, X_temp, y_train, y_temp = train_test_split(X, y, test_size=0.5, random_state=42)
X_val, X_test, y_val, y_test = train_test_split(X_temp, y_temp, test_size=0.5, random_state=42)

alphas = [0.001, 0.01, 0.1, 1, 10, 100]  # You can adjust this list based on your expectations or requirements
best_alpha = None
best_loglikelihood = float('-inf')

for alpha in alphas:
    logit_model = sm.MNLogit(y_train, sm.add_constant(X_train), method='lbfgs')
    result = logit_model.fit_regularized(method='l1', alpha=alpha)
    
    # Use log-likelihood as an example metric
    loglikelihood_val = result.llf
    
    if loglikelihood_val > best_loglikelihood:
        best_loglikelihood = loglikelihood_val
        best_alpha = alpha

print(f"Best alpha: {best_alpha}")

# Once you have the best alpha, you can retrain your model on the combined training and validation set
X_combined = pd.concat([X_train, X_val])
y_combined = pd.concat([y_train, y_val])
logit_model_final = sm.MNLogit(y_combined, sm.add_constant(X_combined), method='lbfgs')
result_final = logit_model_final.fit_regularized(method='l1', alpha=best_alpha)

# Now you can evaluate your model on the test set
# ...
